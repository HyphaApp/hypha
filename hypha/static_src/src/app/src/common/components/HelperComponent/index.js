import * as React from "react";
import PropTypes from 'prop-types';


const HelperComponent = props => {
  return <>
    {props.text && <p className="form__help" dangerouslySetInnerHTML={{ __html: props.text }} />}
    {props.link && <p className="form__help-link">
      <a href={props.link} target="_blank" rel="noopener noreferrer">See help guide for more information.
        <svg className="form__open-icon"><use href="#open-in-new-tab"></use></svg>
      </a>
    </p>}
  </>
}
HelperComponent.propTypes = {
  text: PropTypes.string,
  link: PropTypes.node
}

HelperComponent.displayName = 'HelperComponent';
export default HelperComponent;
 