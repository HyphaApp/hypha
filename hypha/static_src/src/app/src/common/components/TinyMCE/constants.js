export const API_KEY = "fufz3rqdwl1zs7gkxw5wou2g7t00i0eijtyzvy3bgklon5zv";
