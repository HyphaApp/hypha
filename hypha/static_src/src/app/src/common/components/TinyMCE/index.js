import * as React from "react";

import { Editor } from '@tinymce/tinymce-react';
import HelperComponent from "@common/components/HelperComponent";
import PropTypes from 'prop-types';


const TinyMCE = props => {
  return <div className="form__group ">
    <label htmlFor={props.name} className="form__question form__question--choice_field select" required="">
      <span>{props.label} </span>
      {props.required ? <span className="form__required"> *</span> : ""}
    </label>
    <HelperComponent {...props.helperProps} />
    <Editor
      initialValue={props.value}
      init={{
        ...(props.mce_attrs),
        menubar: false
      }}
      onChange={e => props.onChange(props.name, e.level.content)}
      id={props.name}
    />
  </div>
}
TinyMCE.propTypes = {
  label: PropTypes.string,
  required: PropTypes.bool,
  onChange: PropTypes.func,
  value: PropTypes.node,
  helperProps: PropTypes.object,
  name: PropTypes.string,
  mce_attrs: PropTypes.node
}

TinyMCE.displayName = 'TinyMCE';
export default TinyMCE;
