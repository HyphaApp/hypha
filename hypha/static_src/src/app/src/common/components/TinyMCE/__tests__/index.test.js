import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";
import TinyMCE from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });
import { Editor } from '@tinymce/tinymce-react';
import HelperComponent from "@common/components/HelperComponent";


describe("Test tinymce component", () => {
  const text = "test name";
  const name = "test name";
  const label = "test label";
  const required = false;
  const helperProps = {};
  const value = "abc";
  const init = { "a" : 1}
  const onChange = jest.fn();

  const subject = shallow(<TinyMCE
    init = {init}
    text={text}
    name={name}
    label={label}
    required={required}
    value={value}
    helperProps={helperProps}
    onChange={onChange}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have select and span element with passed text", () => {
    expect(subject.find('.form__required').length).toBe(0);
    expect(subject.children().length).toEqual(3);
    expect(subject.find('span').first().text()).toBe(label);
    expect(subject.containsMatchingElement(<Editor />)).toEqual(true);
    expect(subject.containsMatchingElement(<HelperComponent />)).toEqual(true);
    expect(subject.find('Editor').length).toEqual(1);
    subject.find('Editor').props().onEditorChange(1)
    expect(onChange).toHaveBeenCalled();
    expect(onChange).toHaveBeenCalledWith(name, 1)
  });

  test('render a TinyMCE component', () => {
    expect(subject).toMatchSnapshot();
  });

});

describe("Test tinymce component with required", () => {
  const text = "test name";
  const name = "test name";
  const label = "test label";
  const required = true;
  const helperProps = {};
  const value = "abc";
  const init = { "a" : 1}
  const onChange = jest.fn();

  const subject = shallow(<TinyMCE
    init = {init}
    text={text}
    name={name}
    label={label}
    required={required}
    value={value}
    helperProps={helperProps}
    onChange={onChange}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have select and span element with passed text", () => {
    expect(subject.find('.form__required').length).toBe(1);
  });

  test('render a TinyMCE component with required', () => {
    expect(subject).toMatchSnapshot();
  });

});
