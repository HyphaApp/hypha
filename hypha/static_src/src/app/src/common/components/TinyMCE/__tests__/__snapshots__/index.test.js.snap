// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test tinymce component render a TinyMCE component 1`] = `
<div
  className="form__group "
>
  <label
    className="form__question form__question--choice_field select"
    htmlFor="test name"
    required=""
  >
    <span>
      test label
    </span>
  </label>
  <HelperComponent />
  <Editor
    cloudChannel="5"
    id="test name"
    init={
      Object {
        "a": 1,
        "menubar": false,
      }
    }
    initialValue="abc"
    onEditorChange={[Function]}
  />
</div>
`;

exports[`Test tinymce component with required render a TinyMCE component with required 1`] = `
<div
  className="form__group "
>
  <label
    className="form__question form__question--choice_field select"
    htmlFor="test name"
    required=""
  >
    <span>
      test label
    </span>
    <span
      className="form__required"
    >
       *
    </span>
  </label>
  <HelperComponent />
  <Editor
    cloudChannel="5"
    id="test name"
    init={
      Object {
        "a": 1,
        "menubar": false,
      }
    }
    initialValue="abc"
    onEditorChange={[Function]}
  />
</div>
`;
